### ISSUE TITLE
`unittest.mock.Mock()` no longer interpreted as `typing.Any`

### ISSUE THREAD

**Issue by thesadru** [microsoft/pyright#3611]
`unittest.mock.Mock()` can no longer match `type[...]` 

```py
import unittest.mock


class Foo:
    ...


def function(cls: type[Foo]):
    ...


def test_function():
    assert function(unittest.mock.Mock())
```

```
test.py
  test.py:13:21 - error: Argument of type "Mock" cannot be assigned to parameter "cls" of type "Type[Foo]" in function "function"
    Type "Mock" cannot be assigned to type "Type[Foo]" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 information
```

**Comment by erictraut** [comment]
When you say "no longer", did this work at some point in the past with pyright? I tried going back six versions, and I'm seeing the same error.

The issue is that `Mock()` constructs an instance (an object). It's different from most objects in that its class `Mock` derives from `Any`. Even though the class derives from `Any`, pyright treats the object as an instance rather than a class. That's why it is not seen as being compatible with `type[Foo]`. It is considered compatible with `type[Any]`.

**Comment by thesadru** [comment]
> When you say "no longer", did this work at some point in the past with pyright? I tried going back six versions, and I'm seeing the same error.

I do believe so. I don't remember ever having a problem with `Mock()` being incompatible and it's been a fair while since I've written any tests.

How are mocks supposed to be done with `type[...]` if this is incompatible? `NotImplemented` gives me the same error.

**Comment by thesadru** [comment]
> Were you using an older version of pyright until recently? 

I'm only working from memory here. it might not have ever worked honestly.

> You could use a `cast` or a `# type: ignore` call.

Exactly what I'm doing right now. I suppose I'll just stick to that. Thanks.
